assets/css/lesson.scss

```diff
@@ -137,8 +137,10 @@ code {
   background-color: #e7e7e7;
 }
 
-img {
-  max-width: 100%;
+article img {
+    display: block;
```

Comment:
I don't think that we want this one.

Reply:
`display: block` is necessary for the centering to happen.

On a different matter: I think we still want to have `img { max-width: 100%; }` for all "other" images.

Reply:
Consider

```
$ cat f.md 
![Alt text](image.jpg)

Paragraph with ![Alt text](image.jpg) some image.
$ kramdown f.md
<p><img src="image.jpg" alt="Alt text" /></p>

<p>Paragraph with <img src="image.jpg" alt="Alt text" /> some image.</p>
```

I would say that we agree that the first image should be a block without exception. The second image doesn't necessary should be a block, for example, it is a small image of the icon that the reader should click in.

Reply:
True, but I don't see the second type of usage in SWC/DC repos. Is there a need to support it at the moment?

Reply:
> I don't see the second type of usage in SWC/DC repos.

I don't think that we ever use it but I imagine some examples where people might want to use, e.g.

![screenshot from 2018-07-05 18-10-21](https://user-images.githubusercontent.com/1506457/42337458-cb0a9984-807e-11e8-94d3-fc6f29bca07e.png)

> Is there a need to support it at the moment?

I want to avoid the problem of one contributor that is not familiar to CSS who might write

```
Paragraph with ![Alt text](image.jpg) some image.
```

expecting to have the image inline because (1) this is how word processors works and (2) this is the default behaviour in HTML but our lesson have the image as a block.

[Looks like](https://stackoverflow.com/questions/22179642/css-first-child-selector-including-text-nodes) that is impossible to have a CSS-only selector for paragraphs with only one image **and** not text content.

For the record, I suggested https://github.com/carpentries/styles/issues/161 so that we could have better images.

Reply:
My CSS skills aren't great but couldn't we add:

```css
article p > img {
  block: inherit
}
```

to cover the case of images embedded with `<p>` tags?